File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/models/user/renew/RenewLoansResponseEntityMapper.java
package com.hgianastasio.biblioulbrav2.data.models.user.renew;

import com.hgianastasio.biblioulbrav2.core.base.boundaries.Mapper;
import com.hgianastasio.biblioulbrav2.core.user.models.RenewLoansResponse;

import javax.inject.Inject;

/**
 * Created by heitorgianastasio on 4/28/17.
 */

public class RenewLoansResponseEntityMapper extends Mapper<RenewLoansResponse,RenewLoansResponseEntity> {
    @Inject public RenewLoansResponseEntityMapper() {}

    @Override
    public RenewLoansResponseEntity transform(RenewLoansResponse input) {
        return new RenewLoansResponseEntity(input.getMessage(),input.getStatus()?"sim":"nao");
    }

    @Override
    public RenewLoansResponse transformBack(RenewLoansResponseEntity input) {
        return new RenewLoansResponse(input.getStatus().equalsIgnoreCase("sim"),input.getMessage());
    }
}

File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/models/user/renew/RenewLoansResponseEntityMapper.kt
package com.hgianastasio.biblioulbrav2.data.models.user.renew

import com.hgianastasio.biblioulbrav2.core.base.boundaries.Mapper
import com.hgianastasio.biblioulbrav2.core.user.models.RenewLoansResponse
/**
 * Created by heitorgianastasio on 4/28/17.
 */
class RenewLoansResponseEntityMapper constructor() : Mapper<RenewLoansResponse?, RenewLoansResponseEntity?>() {
    override fun transform(input: RenewLoansResponse?): RenewLoansResponseEntity {
        return RenewLoansResponseEntity(input?.message, if (input?.status == true) "sim" else "nao")
    }

    override fun transformBack(input: RenewLoansResponseEntity?): RenewLoansResponse {
        return RenewLoansResponse(input?.status.equals("sim", ignoreCase = true), input?.message)
    }
}